File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> myMap = new HashMap<>();
        myMap.put('I', 1);
        myMap.put('V', 5);
        myMap.put('X', 10);
        myMap.put('L', 50);
        myMap.put('C', 100);
        myMap.put('D', 500);
        myMap.put('M', 1000);

        int result = myMap.get(s.charAt(s.length() - 1));
        for (int i = s.length() - 2; i >= 0; --i) {
            if (myMap.get(s.charAt(i)) < myMap.get(s.charAt(i + 1))) {
                result -= myMap.get(s.charAt(i));
            } else {
                result += myMap.get(s.charAt(i));
            }
        }
        return result;
    }
}

File: 0013-roman-to-integer/0013-roman-to-integer.dart
class Solution {
  int romanToInt(String s) {
    Map<String, int> myMap = {
      'I': 1,
      'V': 5,
      'X': 10,
      'L': 50,
      'C': 100,
      'D': 500,
      'M': 1000,
    };

    int result = myMap[s[s.length - 1]]!;
    for (int i = s.length - 2; i >= 0; --i) {
      if (myMap[s[i]]! < myMap[s[i + 1]]!) {
        result -= myMap[s[i]]!;
      } else {
        result += myMap[s[i]]!;
      }
    }
    return result;
  }
}
